src/Menu/index.tsx

```diff
@@ -108,23 +98,46 @@ function isReferenceObject(reference: any): reference is ReferenceElement {
  */
 export const Menu: React.FC<Props> = ({
   children,
+  closeOnMenuItemClick = true,
   iconSize,
-  instanceRef,
   scrollableContent = false,
+  content,
   ...props
 }) => {
+  const instanceRef = React.useRef<
+    Parameters<
+      NonNullable<React.ComponentProps<typeof AbstractTooltip>["onCreate"]>
+    >[0]
+  >();
+  const inheritedMenuItemClickListener = useMenuItemClickListener();
   const inheritedIconSize = useMenuIconSize();
 
+  const onClick = React.useCallback<React.MouseEventHandler>(
+    event => {
+      if (!closeOnMenuItemClick) return;
```

Comment:
If the intention is to always forward the click to a potential parent menu, shouldn't we only be making the `hide` function below execute conditionally?  E.g. get rid of the early return, and
```
  if (inheritedMenuItemClickListener) {
    inheritedMenuItemClickListener(event);
  }
  if (closeOnMenuItemClick) {
    instanceRef.current?.hide();
  }
```

Reply:
Yes, this would totally bite us for nested menus. Good catch.

Reply:
Done!